packages/salesforcedx-vscode-core/test/jest/decorators/scratch-org-decorator.test.ts

```diff
@@ -0,0 +1,66 @@
+import { ConfigUtil } from '@salesforce/salesforcedx-utils-vscode';
+import * as vscode from 'vscode';
+import { ORG_OPEN_COMMAND } from '../../../src';
+import { showOrg } from '../../../src/decorators';
+import { nls } from '../../../src/messages';
+
+describe('scratch org decorator', () => {
+  const testUser = 'test@username.com';
+  const browserIcon = `$(browser)`;
+  const openOrgCommand = ORG_OPEN_COMMAND;
+  let createFileSystemWatcherMock: jest.SpyInstance;
+  let getDefaultUsernameOrAliasMock: jest.SpyInstance;
+  let createStatusBarItemMock: jest.SpyInstance;
+  const mockStatusBarItem: any = {};
+  let mockWatcher: any;
+
+  beforeEach(() => {
+    mockStatusBarItem.tooltip = '';
+    mockStatusBarItem.command = '';
+    mockStatusBarItem.text = '';
+    mockStatusBarItem.show = jest.fn();
+    mockStatusBarItem.dispose = jest.fn();
+    mockWatcher = {
+      onDidChange: jest.fn(),
+      onDidCreate: jest.fn()
+    };
+    createStatusBarItemMock = (vscode.window.createStatusBarItem as any).mockReturnValue(
+      mockStatusBarItem
+    );
+    createFileSystemWatcherMock = (vscode.workspace
+      .createFileSystemWatcher as any).mockReturnValue(mockWatcher);
+    getDefaultUsernameOrAliasMock = jest
+      .spyOn(ConfigUtil, 'getDefaultUsernameOrAlias');
+  });
+
+  describe('show Org', () => {
+    it('should show the browser icon in the status bar when a default username is set', async () => {
+      getDefaultUsernameOrAliasMock.mockResolvedValue(testUser);
+      await showOrg();
+      expect(getDefaultUsernameOrAliasMock).toHaveBeenCalled();
+      expect(createStatusBarItemMock).toHaveBeenCalled();
+      expect(mockStatusBarItem.tooltip).toEqual(nls.localize('status_bar_open_org_tooltip'));
+      expect(mockStatusBarItem.command).toEqual(openOrgCommand);
+      expect(mockStatusBarItem.show).toHaveBeenCalled();
+      expect(mockStatusBarItem.text).toEqual(browserIcon);
+    });
+    it('should not show the browser icon in the status bar when a default username is not set', async () => {
+      await showOrg();
+      expect(getDefaultUsernameOrAliasMock).toHaveBeenCalled();
+      expect(createStatusBarItemMock).not.toHaveBeenCalled();
+      expect(mockStatusBarItem.tooltip).toEqual('');
+      expect(mockStatusBarItem.command).toEqual('');
+      expect(mockStatusBarItem.show).not.toHaveBeenCalled();
+      expect(mockStatusBarItem.text).toEqual('');
+    });
+    it('should dispose and set to undefined the status bar when a default username is not set and the status bar exists', async () => {
+      mockStatusBarItem.text = 'text';
+      getDefaultUsernameOrAliasMock.mockResolvedValue(undefined);
+      await showOrg();
+      expect(getDefaultUsernameOrAliasMock).toHaveBeenCalled();
+      expect(createStatusBarItemMock).not.toHaveBeenCalled();
+      expect(mockStatusBarItem.dispose).toHaveBeenCalled();
```

Comment:
The status bar item hasn't been created yet in this test, so dispose won't be called here.  In order to test that dispose was called on the internal instance of statusBarItem, we'll have to call showOrg legitimately first to create the statusBarItem and have a handle on the mock:

<img width="500" alt="2022-11-11_14-31-05" src="https://user-images.githubusercontent.com/46458081/201440081-45e038a2-cb96-4c1a-bd4b-7c43f605ee14.png">